src/lib/workspace/DownloadTab/hooks/useGetReleaseFiles.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect } from 'react';
+
+// Utilities
+import { DownloadClient } from '../../../core/api/DownloadClient';
+
+// Definitions
+import { DownloadTabStudyRelease } from '../types';
+
+export type ReleaseFile = {
+  fileName: string;
+  fileSize: string;
+  fileType: string;
+  fileDescription: string;
+  fileUrl: string | undefined;
+};
+
+/**
+ * Obtain information from the Download service about the
+ * files that are available to download for a given release.
+ */
+export function useGetReleaseFiles(
+  studyId: string,
+  release: DownloadTabStudyRelease,
+  downloadClient: DownloadClient,
+  setReleaseFiles: React.Dispatch<React.SetStateAction<Array<ReleaseFile>>>
+) {
+  useEffect(() => {
```

Comment:
You might be able to get rid of some boilerplate by using [usePromise](https://github.com/VEuPathDB/web-eda/blob/28df6270b12d854637258833f9bb0d8a9655db45/src/lib/core/hooks/promise.ts) instead of `useState` / `useEffect`.

In `useGetReleaseFiles`, you'd:

1. Initialize a `fetchReleaseFiles` callback (memoized via a `useCallback` with `studyId`, `release.downloadServiceReleaseId`, and `downloadClient` dependencies)
2. Return `usePromise(fetchReleaseFiles)`

In the interest of merging sooner rather than later, we can leave this as "future work."